logspout/logspout.go

```diff
@@ -65,32 +70,42 @@ func syslogStreamer(target Target, types []string, logstream chan *Log) {
 			continue
 		}
 		tag, pid, data := getLogParts(logline)
-		var conn net.Conn
+
+		// HACK: Go's syslog package hardcodes the log format, so let's send our own message
+		data = fmt.Sprintf("%s %s[%s]: %s",
+			time.Now().Format(getopt("DATETIME_FORMAT", dtime.DeisDatetimeFormat)),
+			tag,
+			pid,
+			data)
+
 		if strings.EqualFold(target.Protocol, "tcp") {
 			addr, err := net.ResolveTCPAddr("tcp", target.Addr)
 			assert(err, "syslog")
-			tcpconn, err := net.DialTCP("tcp", nil, addr)
+			conn, err := net.DialTCP("tcp", nil, addr)
+			assert(err, "syslog")
+			assert(conn.SetWriteBuffer(MAX_TCP_MSG_BYTES), "syslog")
```

Comment:
[`assert`](https://golang.org/doc/faq#assertions) does seem like an anti-pattern in Go, but the code in question was already using it heavily so this doesn't really change existing error handling.